static-instrumenter/maven-plugin/src/main/java/io/opentelemetry/contrib/staticinstrumenter/plugin/maven/PackagingSupportFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter.plugin.maven;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+public final class PackagingSupportFactory {
+
+  private static final Map<String, Supplier<PackagingSupport>> SUPPORTED_FRAMEWORKS =
+      Map.of(
+          "BOOT-INF/classes/", () -> new PackagingSupport("BOOT-INF/classes/"),
+          "WEB-INF/classes/", () -> new PackagingSupport("WEB-INF/classes/"));
```

Comment:
Nit: the value is a simple transformation of the key, this could be just a set of framework classes directory paths and you could just call `new PackagingSupport(...)` in `packagingSupportFor()`

Reply:
True that.